### PR TITLE
Several fixes and improvements for code reading

### PR DESCRIPTION
**Pull request by NeonKnightOA**

### CODE REVIEW

File: code/game/g_local.h [hunk 1]
```diff
@@ -1431,4 +1431,5 @@ qboolean G_IsATeamGametype(int check);	/* Whether the gametype is team-based or
 qboolean G_UsesTeamFlags(int check);	/* Whether the gametype uses the red and blue flags. */
 qboolean G_UsesTheWhiteFlag(int check);	/* Whether the gametype uses the neutral flag. */
 qboolean G_IsARoundBasedGametype(int check);	/* Whether the gametype uses the neutral flag. */
+qboolean G_HasEliminationRules(int check);	/* Whether the game uses Elimination rules (all weapons + no pickups). */
```
Comment (sago007) [inline code/game/g_local.h]: Drop the parameter and just check for g_gametype.integer inside the function. Having an argument signals to the reader that the function does not read cvars directly and that is false.

Comment (NeonKnightOA) [reply code/game/g_local.h]: I'll remove the HER commit from this branch (it's also related to #91) and make it into a separate branch so we can polish it from there. What about the rest of the commits, are they OK?

Comment (NeonKnightOA) [reply code/game/g_local.h]: Done. I'll open a separate PR for that one.

**Comment by NeonKnightOA** [comment]
On second thought, I should better prepare a proper code reading fix before anything else.

File: code/game/g_local.h (hunk 1 above)
Comment (NeonKnightOA) [reply code/game/g_local.h]: There, the function has been moved to #111.

**Pull request closed without merging**